integrations/Qualaroo/browser.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import logger from "../../utils/logUtil";
+import { NAME } from "./constants";
+import ScriptLoader from "../ScriptLoader";
+import { recordQualarooEvents, transformUserTraits } from "./utils";
+
+class Qualaroo {
+  constructor(config, analytics) {
+    if (analytics.logLevel) {
+      logger.setLogLevel(analytics.logLevel);
+    }
+    this.name = NAME;
+    this.customerId = config.customerId;
+    this.siteToken = config.siteToken;
+    this.recordQualarooEvents = config.recordQualarooEvents;
+    this.eventsToStandard = config.eventsToStandard;
+    this.updateEventNames = config.updateEventNames;
+    this.eventsList = config.eventsList;
+  }
+
+  loadScript() {
+    window._kiq = window._kiq || [];
+    ScriptLoader(
+      "qualaroo-integration",
+      `https://cl.qualaroo.com/ki.js/${this.customerId}/${this.siteToken}.js`
+    );
+  }
+
+  init() {
+    logger.debug("===In init Qualaroo===");
+    this.loadScript();
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Qualaroo===");
+    return !!window._kiq && typeof window._kiq === "object";
+  }
+
+  isReady() {
+    logger.debug("===In isReady Qualaroo===");
+    if (this.recordQualarooEvents) {
+      recordQualarooEvents(
+        this.updateEventNames,
+        this.eventsList,
+        this.eventsToStandard
+      );
+    }
+    return !!window._kiq && typeof window._kiq === "object";
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Qualaroo identify===");
+    const { message } = rudderElement;
+    const userId =
+      get(message, "context.traits.email") ||
```

Comment:
Please do not use get function while developing web device mode. Let's change get to dot (.) operator with optional chaining.

Reply:
any specific reason?. We have used get function in most of our existing integrations.

Reply:
yes, we have decided this because get, set are some expensive functions, that does decrease the overall efficiency of the SDK

Reply:
Code updated. Have used optional chaining.